openmdao/core/component.py

```diff
@@ -1259,6 +1259,10 @@ def _declare_partials(self, of, wrt, dct):
                 meta['cols'] = cols
                 meta['shape'] = shape = (abs2meta[abs_key[0]]['size'], abs2meta[abs_key[1]]['size'])
 
+                if abs2meta[abs_key[0]]['size'] == 0:
+                    msg = '{}: Is an array of size 0'
```

Comment:
msg should include the name of the variable that is size 0  (abs_key[0] in this case). Also, is there a reason we're not checking if the variable corresponding to abs_key[1] is size 0 as well?